BedrockServer.cpp

```diff
@@ -2256,45 +2259,47 @@ void BedrockServer::handleSocket(Socket&& socket, bool fromControlPort, bool fro
                         // command that was already responded to in `buildCommandFromRequest` and we can move on to the
                         // next thing immediately.
                         mutex m;
-
-                        // Defer locking until we actually have to.
-                        unique_lock<mutex> lock(m, defer_lock);
                         condition_variable cv;
+                        atomic<bool> finished = false;
 
-                        function<void()> callback = [&m, &cv]() {
+                        function<void()> callback = [&m, &cv, &finished]() {
                             // Lock the mutex above (which will be locked by this thread while we're queuing), which waits
                             // for `handleSocket` to release it's lock (by calling `wait`), and then notify the waiting
                             // socket thread.
-                            lock_guard lock(m);
+                            lock_guard l(m);
```

Comment:
Is the variable rename necessary? Otherwise I would leave it to A) Reduce the diff; and B) Make it more readable. NAB though.

Reply:
It's not, this is just an artifact leftover from rearranging this block several times with different variations on it. I can change it back.